src/main/scala/cromwell/engine/backend/local/LocalEngineFunctions.scala

```diff
@@ -96,9 +112,37 @@ class LocalEngineFunctions(cwd: Path, stdout: Path, stderr: Path) extends LocalE
       singleArgument <- extractSingleArgument(params)
       if singleArgument.wdlType.isInstanceOf[WdlMapType]
       tsvSerialized <- singleArgument.asInstanceOf[WdlMap].tsvSerialize
-      (path, writer) = FileUtil.tempFileAndWriter("map", cwd.toFile)
-      _ <- Try(writer.write(tsvSerialized))
-      _ <- Success(writer.close())
-    } yield WdlFile(path.toAbsolutePath.toString)
+      file <- writeContent("map", tsvSerialized)
+    } yield file
+  }
+
+  override protected def write_object(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      if singleArgument.wdlType == WdlObjectType
+      tsvSerialized <- singleArgument.asInstanceOf[WdlObject].tsvSerialize
+      file <- writeContent("object", tsvSerialized)
+    } yield file
+  }
+
+  override protected def write_objects(params: Seq[Try[WdlValue]]): Try[WdlFile] = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      if singleArgument.wdlType.isAnArrayOf(WdlObjectType)
+      tsvSerialized <- singleArgument.asInstanceOf[WdlArray].tsvSerialize
+      file <- writeContent("array", tsvSerialized)
+    } yield file
+  }
+
+  protected def writeContent(baseName: String, content: String) = {
+    val (path, writer) = FileUtil.tempFileAndWriter("array", cwd.toFile)
```

Comment:
Is this going to clash if the same task writes two arrays?

e.g. task x {
 ...
 ...
 output {
  File a = write_array(blah)
  File b = write_array(blah2)
 }
}